crates/cargo-wdk/src/actions/new/mod.rs

```diff
@@ -291,3 +291,476 @@ impl<'a> NewAction<'a> {
+            (Verbosity::new(0, 1), Some("-q".to_string())), // Quiet
+            (Verbosity::new(1, 0), Some("-v".to_string())), // Verbose
+        ];
+
+        for (verbosity_level, expected_flag) in cases {
+            let test_new_action = TestNewAction::new(path)
+                .expect_cargo_new(None, expected_flag)
+                .expect_copy_lib_rs_template(true)
+                .expect_update_cargo_toml(true, true, true)
+                .expect_create_inx_file(true)
+                .expect_copy_build_rs_template(true)
+                .expect_copy_cargo_config(true);
+            let result = NewAction::new(
+                path,
+                driver_type,
+                verbosity_level,
+                &test_new_action.mock_exec,
+                &test_new_action.mock_fs,
+            )
+            .run();
+            assert!(result.is_ok());
+        }
+    }
+
+    #[test]
+    fn when_cargo_new_fails_then_returns_cargo_new_command_error() {
+        let path = Path::new("test_driver_fail_cargo_new");
+        let driver_type = DriverType::Kmdf;
+        let verbosity_level = Verbosity::default();
+
+        // Set up mocks with failure at cargo new step
+        let test_new_action = TestNewAction::new(path).expect_cargo_new(
+            Some(Output {
+                status: ExitStatus::from_raw(1),
+                stdout: vec![],
+                stderr: "some error".into(),
+            }),
+            None,
+        );
+
+        let result = NewAction::new(
+            path,
+            driver_type,
+            verbosity_level,
+            &test_new_action.mock_exec,
+            &test_new_action.mock_fs,
+        )
+        .run();
+        assert!(
+            matches!(result, Err(NewActionError::CargoNewCommand(_))),
+            "Expected CargoNewCommand error"
+        );
+    }
+
+    #[test]
+    fn when_copy_lib_rs_template_fails_then_returns_filesystem_error() {
+        let path = Path::new("test_driver_fail_lib");
+        let driver_type = DriverType::Kmdf;
+        let verbosity_level = Verbosity::default();
+
+        // Set up mocks with failure at copy lib rs template to driver project step
+        let test_new_action = TestNewAction::new(path)
+            .expect_cargo_new(None, None)
+            .expect_copy_lib_rs_template(false); // Force failure here
+
+        let result = NewAction::new(
+            path,
+            driver_type,
+            verbosity_level,
+            &test_new_action.mock_exec,
+            &test_new_action.mock_fs,
+        )
+        .run();
+        assert!(
+            matches!(
+                result,
+                Err(NewActionError::FileSystem(FileError::WriteError(_, _)))
+            ),
+            "Expected FileSystem WriteError from copy_lib_rs_template"
+        );
+    }
+
+    #[test]
+    fn when_update_cargo_toml_fails_for_multiple_cases_then_returns_filesystem_error() {
```

Comment:
```suggestion
    fn when_update_cargo_toml_fails_then_returns_filesystem_error() {
```

`for_multiple_cases` doesn't add anything useful